lute/static/js/dict-tabs.js

```diff
@@ -1,22 +1,61 @@
 "use strict";
 
+let dictTabs = [];
+
+class DictTab {
+  constructor(dict, frameName) {
+    const dictInfo = dict ? getDictInfo(dict) : null;
+    this.dictID = dictInfo ? dictInfo.id : null;
+
+    this.frame = this.createIFrame(frameName);
+    this.btn = document.createElement("button");
+    this.btn.classList.add("dict-btn");
+
+    if (this.dictID != null) {
+      this.label = dictInfo.label;
+      this.isExternal = dictInfo.isExternal;
+      this.btn.dataset.dictId = this.dictID;
```

Comment:
With the class in place, it may be possible to get rid of the `dataset` stuff, and just have the information stored in the DictTab class.  This applies to the button dataset as well as the frame, eg instead of `dictTabs.forEach(tab => tab.frame.dataset.contentLoaded = "false");` you might be able to say something like `dictTabs.forEach(tab => tab.reset());` or similar.